### PR TITLE
fix: append existing PYTHONPATH to CMAKE_PYTHON_PATH

### PR DESCRIPTION
**Pull request by wdconinc**
This addresses #3500. It approaches it slightly differently than the code included there, in order to behave correctly when `PYTHONPATH` is not set and to remain within the CMake version 3.6 minimum (prevents `JOIN`).

This now installs correctly in my environment where `pip` is installed in a different prefix from `python` (and this prefix is included in `PYTHONPATH`).

### CODE REVIEW

**Comment by wdconinc** [comment]
> mlpack.mlpack (macOS Python)  Failing after 2m — macOS Python failed

This seems to be due to a missing `pandas`, per https://dev.azure.com/mlpack/mlpack/_build/results?buildId=9260&view=logs&jobId=78182f4f-e247-5271-bde3-6071f836eaa6&j=78182f4f-e247-5271-bde3-6071f836eaa6&t=7738125e-f354-5cc7-1ce8-930b5c7ee4b5

File: src/mlpack/bindings/python/CMakeLists.txt [hunk 1]
```diff
@@ -203,6 +203,9 @@ execute_process(COMMAND ${PYTHON_EXECUTABLE}
     "${PYTHON_INSTALL_PREFIX}"
     OUTPUT_VARIABLE CMAKE_PYTHON_PATH)
 string(STRIP "${CMAKE_PYTHON_PATH}" CMAKE_PYTHON_PATH)
+if (DEFINED ENV{PYTHONPATH})
+    string(APPEND CMAKE_PYTHON_PATH : $ENV{PYTHONPATH})
```
Comment (rcurtin) [inline src/mlpack/bindings/python/CMakeLists.txt:207]:
```suggestion
  string(APPEND CMAKE_PYTHON_PATH : $ENV{PYTHONPATH})
```
Tiny nit, two-space tabs. :)

**Review by rcurtin — APPROVED** [review]
Everything looks good to me here.  Thank you for opening the PR!  I'll debug the OS X issue separately.  It's strange: the build logs indicate that pandas did get installed through pip, but then somehow it is not found by CMake?  Not looking forward to the deep dive...

**Comment by rcurtin** [comment]
Thanks for the fix! :+1:

**Pull request merged**